recipes/Libri-Light/self-supervised-learning/make_librilight_csv.py

```diff
@@ -0,0 +1,168 @@
+        waveform, sample_rate = torchaudio.load(str(flac_file))
+        num_frames = waveform.size(1)
+        duration_seconds = num_frames / sample_rate
+        if duration_seconds > max_length:
+            continue
+        audio_length_seconds = waveform.shape[1] / sample_rate
+        csv_file = f"{csv_file_folder}/{flac_file.parent.stem}.csv"
+        with open(csv_file, mode="a", newline="", encoding="utf-8") as csvfile:
+            csv_writer = csv.writer(
+                csvfile, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [flac_file_name, audio_length_seconds, str(flac_file)]
+            )
+
+
+def processes_folder(
+    data_path: str,
+    csv_file_folder: str,
+    max_length: float,
+    n_processes: int,
+) -> None:
+    """Prepare the csv files for one of the Libri-Light splits
+
+    Arguments
+    ---------
+    data_path : str
+        Path to the Libri-Light split after vad
+    csv_file_folder: str
+        Path to the output folder of generated csv files
+    max_length: float
+        The max length (seconds) for each prepared audio clip
+    n_processes: int
+        Number of parallel processes
+    """
+    print("Processing each subfolder of this split")
+    os.makedirs(csv_file_folder, exist_ok=True)
+    os.makedirs(f"{csv_file_folder}/tmp", exist_ok=True)
+    list_dir = pathlib.Path(data_path)
+    tasks = []
+    for x in list_dir.iterdir():
+        tasks.append((x, f"{csv_file_folder}/tmp", max_length))
+    with multiprocessing.Pool(processes=n_processes) as pool:
+        for _ in tqdm.tqdm(
+            pool.imap_unordered(make_csv_for_each, tasks), total=len(tasks)
+        ):
+            pass
+
+
+def merge_csv_files(
+    csv_file_folder: str,
+) -> None:
+    """Merges the csv files for each subfolder into one single file
+
+    Arguments
+    ---------
+    csv_file_folder: str
+        Path to the output folder of generated csv files
+    """
+    print("Merging the csvs of each subfolder into one csv")
+    file_list = [str(x) for x in Path(f"{csv_file_folder}/tmp").glob("*.csv")]
+    output_file = f"{csv_file_folder}/train.csv"
+    fieldnames = ["ID", "duration", "wav"]
+
+    with open(output_file, mode="a", newline="", encoding="utf-8") as outfile:
+        csv_writer = csv.writer(
+            outfile, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+        )
+        csv_writer.writerow(fieldnames)
+        for file_path in tqdm.tqdm(file_list):
+            with open(file_path, mode="r", encoding="utf-8") as infile:
+                reader = csv.reader(infile)
+
+                # filter out bad rows
+                for row in reader:
+                    if len(row) == 3 and os.path.exists(row[-1]):
+                        new_row = [row[-1], row[1], row[2]]
+                        csv_writer.writerow(new_row)
+                    else:
+                        print(f"bad row {row}")
```

Comment:
can you add in the docstring of the function what does it means to have "bad rows" ?